projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,286 @@
+            this.el.classList.remove('dot-dragover');
+            this.errorMessage = '';
+            const droppedFile: File = evt.dataTransfer.files[0];
+            if (isFileAllowed(droppedFile.name, this.allowedFileTypes)) {
+                this.setValue(droppedFile);
+                this.setFileDisplayName(droppedFile.name);
+            } else {
+                this.errorMessage = 'validationMessage';
+                this.setValue(null);
+            }
+        }
+    }
+
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid(), this.required)
+        };
+    }
+
+    render() {
+        return (
+            <Fragment>
+                <dot-label
+                    label={this.label}
+                    required={this.required}
+                    name={this.name}
+                    tabindex="0"
+                >
+                    <dot-binary-text-field
+                        placeholder={this.placeholder}
+                        required={this.required}
+                        disabled={this.disabled}
+                        value={this.binaryTextValue}
+                        accept={this.allowedFileTypes.join()}
+                    />
+                    <input
+                        aria-describedby={getHintId(this.hint)}
+                        class={getErrorClass(this.status.dotValid)}
+                        disabled={this.disabled}
+                        id={getId(this.name)}
+                        onChange={(event: Event) => this.fileChangeHandler(event)}
+                        required={this.required || null}
+                        type="file"
+                        accept={this.accept}
+                        value={this.value}
+                    />
+                    <button disabled={this.disabled} onClick={() => this.buttonHandler()}>
+                        {this.buttonLabel}
+                    </button>
+                </dot-label>
+                {getTagHint(this.hint)}
+                {getTagError(this.shouldShowErrorMessage(), this.getErrorMessage())}
+            </Fragment>
+        );
+    }
+
+    private fileChangeHandler(event: Event): void {
+        const input: HTMLInputElement = event.target as HTMLInputElement;
+        this.errorMessage = '';
+        if (isFileAllowed(input.files[0].name, this.allowedFileTypes)) {
+            this.setValue(input.files[0]);
+            this.setFileDisplayName(input.files[0].name);
+        } else {
+            event.preventDefault();
+            this.errorMessage = 'validationMessage';
+            this.setValue(null);
+            this.setFileDisplayName('');
+        }
+    }
+
+    private buttonHandler(): void {
+        const fileInput: HTMLInputElement = this.el.querySelector('dot-label input[type="file"]');
+        fileInput.click();
+    }
+
+    private setFileDisplayName(name: string): void {
+        this.binaryTextValue = name;
```

Comment:
So the method name and the prop name (the one that is setting the value to) are so different. Can we try to match names here? Do we need the method at all?

Reply:
after refactor, not needed.